modules/ravestate_roboy_ad_demo/__init__.py

```diff
@@ -0,0 +1,115 @@
+                                               msg_type=Bool)
+
+        # add the ros2 properties to context
+        ctx.push(ros2_properties_parent.id(), publish_dr_to_ad)
+        ctx.push(ros2_properties_parent.id(), subscribe_ad_to_dr)
+        ctx.push(ros2_properties_parent.id(), publish_handshake_motion)
+        ctx.push(ros2_properties_parent.id(), subscribe_skin_to_dr)
+
+        @state(cond=s("nlp:triples:changed"),
+               read="nlp:triples", write=(publish_dr_to_ad.id(), raw_out.id()))
+        def pickup_requested(ctx: ContextWrapper):
+            # Trigger phrase: "Pick me up."
+            triples = ctx["nlp:triples"]
+            if triples[0].match_either_lemma(pred={"pick"}):
+                ctx[publish_dr_to_ad.id()] = String(data=PICKUP_REQUESTED_MSG)
+                ctx[raw_out.id()] = "I'm on my way!"
+            else:
+                return Resign()
+
+        @state(read=subscribe_ad_to_dr.id(), write=raw_out.id())
+        def arrived_somewhere(ctx: ContextWrapper):
+            # processes message from AD
+            msg = ctx[subscribe_ad_to_dr.id()].data
+            if msg == ARRIVED_AT_PICKUP_POINT_MSG:
+                ctx[raw_out.id()] = "Hey dad! Hop on!"
+            elif msg == ARRIVED_AT_DROPOFF_POINT_MSG:
+                ctx[raw_out.id()] = "We have arrived."
+            else:
+                logger.error(f"Unexpected message: {msg} received on topic: {AD_TO_DR_TOPIC}")
+
+        @state(cond=s("nlp:triples:changed"),
+               read="nlp:triples", write=(publish_dr_to_ad.id(), raw_out.id()))
+        def start_driving(ctx: ContextWrapper):
+            # Trigger phrase: "Start driving." or "Go."
+            triples = ctx["nlp:triples"]
+            if triples[0].match_either_lemma(pred={"drive", "start", "go"}):
+                # lets go
+                ctx[publish_dr_to_ad.id()] = String(data=START_DRIVING_MSG)
+                ctx[raw_out.id()] = "Fasten your seat belt and off we go!"
+            else:
+                return Resign()
+
+        @state(read=subscribe_skin_to_dr.id(), write=raw_out.id())
+        def handshake_detected(ctx: ContextWrapper):
+            msg = ctx[subscribe_skin_to_dr.id()].data
+            if msg:
+                ctx[raw_out.id()] = "Ah. I see you are very good at handshaking."
+            else:
+                return Resign()
+
+        @state(cond=s("stalker:lennart"), write=publish_handshake_motion.id())
+        def lennart_recognized(ctx: ContextWrapper):
+            # check if it is lennart, greets lennart (this is done in stalker) -> TODO signal lennarts-here?
+            # triggers handshake: topic ReplayTrajectory which takes a string as a name
+            ctx[publish_handshake_motion.id()] = String(data=ctx.conf(key=HANDSHAKE_MOTION_NAME_CONFIG))
+
+        for st in [pickup_requested, start_driving, arrived_somewhere, lennart_recognized, handshake_detected]:
+            mod.add(st)
+            ctx.add_state(st)
```

Comment:
This module is a good example on how to use ros2 properties, but should it be in core ravestate?

Reply:
No, we should delete it.